libra-node/src/main_node.rs

```diff
@@ -255,14 +241,15 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> LibraHandle {
 
     let waypoint = config::waypoint(&node_config.base.waypoint);
 
-    if let Some(network) = node_config.validator_network.as_mut() {
+    let (node_config, validator_network) = node_config.split_off_validator_networks_identity();
+    if let Some(validator_network) = validator_network {
         let (runtime, mut network_builder) = setup_network(
-            network,
+            validator_network,
             RoleType::Validator,
             Arc::clone(&db_rw.reader),
             waypoint,
         );
-        let peer_id = network_builder.peer_id();
+        let peer_id = network_builder.peer_id;
 
         // Set up to listen for network configuration changes from StateSync.
         if let Some(conn_mgr_reqs_tx) = network_builder.conn_mgr_reqs_tx() {
```

Comment:
There is a conflict between this code and #4299 .   This set of changes is also unrelated to the announced intent of the PR to make NetworkBuilder take the NetworkConfig.  

Suggest to role this set of changes back and keep the PR narrowly focused on having NetworkBuilder consume NetworkConfig.  Those changes should not have a blast radius that reaches outside of setup_network.